src/core_plugins/timelion/public/directives/timelion_expression_suggestions/timelion_expression_suggestions.html

```diff
@@ -12,7 +12,7 @@
     tabindex="0"
     data-suggestion-list-item
     ng-class="{active: $index === selectedIndex}"
-    ng-repeat="suggestion in suggestions track by $index | orderBy:'name'"
+    ng-repeat="suggestion in suggestions | orderBy:'name'"
```

Comment:
I dug into this a bit more and found a [SO question](https://stackoverflow.com/questions/34175783/filternotarray-expected-array-but-received-0) which is identical to the problem we're solving. I think that the [upgrade to Angular 1.6.5](https://github.com/elastic/kibana/pull/13543) exposed the faulty syntax because it introduced the [error which gets thrown](https://code.angularjs.org/1.6.5/docs/error/orderBy/notarray).

So I'd just suggest retaining the `track by` part of this expression, but put it at the end as the SO answer suggests:

```
ng-repeat="suggestion in suggestions | orderBy:'name' track by $index"
```